### PR TITLE
add configuration for NewRelic insights url

### LINKED ISSUE #4081: Ability to configure New Relic Insights Insert URL in NewRelicEmitter
## What challenge are you facing?

I want to use `EU based New Relic accounts` for concourse metrics. Currently this is not possible due to hardcoded url in the Emitter config.

https://github.com/concourse/concourse/blob/ebd913b0ea88ff204c790db9284c3d849e34f7df/atc/metric/emitter/newrelic.go#L58

You can find EU endpoints for [New Relic Here](https://docs.newrelic.com/docs/using-new-relic/welcome-new-relic/getting-started/introduction-eu-region-data-center#endpoints) 

## What would make this better?

Along with `AccountID` and `APIKey` we can also have Insights Insert API in the New Relic Config with  `https://insights-collector.newrelic.com` as default to be compatible, like

```go
type NewRelicConfig struct {
	AccountID string `long:"newrelic-account-id" description:"New Relic Account ID"`
	APIKey    string `long:"newrelic-api-key" description:"New Relic Insights API Key"`
	Url       string `long:"newrelic-insights-api-url" default:"https://insights-collector.newrelic.com" description:"Base Url for insights Insert API"`
	ServicePrefix string `long:"newrelic-service-prefix" default:"" description:"An optional prefix for emitted New Relic events"`
}
```

https://github.com/concourse/concourse/blob/ebd913b0ea88ff204c790db9284c3d849e34f7df/atc/metric/emitter/newrelic.go#L31

## Are you interested in implementing this yourself?

Yes, Given an opportunity I shall gladly implement this myself.

### PR DESCRIPTION
**Pull request by shyamz-22**
This configuration allows to use EU based NewRelic account.

# Why is this PR needed?

I want to use `EU based New Relic accounts` for concourse metrics. Currently this is not possible due to hardcoded url in the Emitter config.

https://github.com/concourse/concourse/blob/dee3092f98a2a32e00464251057c3b6447d2be7d/atc/metric/emitter/newrelic.go#L68

You can find EU endpoints for [New Relic Here](https://docs.newrelic.com/docs/using-new-relic/welcome-new-relic/getting-started/introduction-eu-region-data-center#endpoints) 

# What is this PR trying to accomplish?

Along with `AccountID` and `APIKey` we can also have Insights Insert API in the New Relic Config with  `https://insights-collector.newrelic.com` as default to be backward compatible

https://github.com/concourse/concourse/blob/dee3092f98a2a32e00464251057c3b6447d2be7d/atc/metric/emitter/newrelic.go#L39

fixes #4081

# How does it accomplish that?

By adding a url configuration that can be used as a base URL, if this is set then this is used else the current default implementation is used

```go
type NewRelicConfig struct {
  AccountID          string        `long:"newrelic-account-id" description:"New Relic Account ID"`
  APIKey             string        `long:"newrelic-api-key" description:"New Relic Insights API Key"`
  Url                string        `long:"newrelic-insights-api-url" default:"https://insights-collector.newrelic.com" description:"Base Url for insights Insert API"`
  ServicePrefix      string        `long:"newrelic-service-prefix" default:"" description:"An optional prefix for emitted New Relic events"`
  BatchSize          uint64        `long:"newrelic-batch-size" default:"2000" description:"Number of events to batch together before emitting"`
  BatchDuration      time.Duration `long:"newrelic-batch-duration" default:"60s" description:"Length of time to wait between emitting until all currently batched events are emitted"`
  DisableCompression bool          `long:"newrelic-batch-disable-compression" description:"Disables compression of the batch before sending it"`
}
```

# Contributor Checklist

> Are the following items included as part of this PR? If no, please say why not.

- [✔] Unit tests
- [ ] Integration tests: I have not found a test for emitters
- [ ] Updated documentation (located at https://github.com/concourse/docs): Not applicable
- [✔ ] Updated release notes (located at https://github.com/concourse/concourse/tree/master/release-notes)

# Reviewer Checklist

> This section is intended for the core maintainers only, to track review progress.

> Please do not fill out this section.

- [ ] Code reviewed
- [ ] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [ ] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the [BOSH](https://github.com/concourse/concourse-bosh-release) 
      and [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for the [integration tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env) (for example, if they are Garden configs that are not displayed in the `--help` text).

### CODE REVIEW

**Comment by xtreme-sameer-vohra** [comment]
Hey @shyamz-22 ,
Thank you for submitting the PR. 
I'll be taking a look at it shortly :)

**Comment by xtreme-sameer-vohra** [comment]
Thank you for adding the tests and release notes 👍

**Comment by shyamz-22** [comment]
If all is well I can make a PR to the Helm chart as well :)

**Comment by shyamz-22** [comment]
@xtreme-sameer-vohra hi I cannot find your review for the url function can you help me find it?

**Comment by xtreme-sameer-vohra** [comment]
> @xtreme-sameer-vohra hi I cannot find your review for the url function can you help me find it?

oops, it appears that I hadn't submitted the review. Just did it now :)

File: atc/metric/emitter/newrelic.go [hunk 1]
```diff
@@ -57,6 +59,13 @@ func (config *NewRelicConfig) IsConfigured() bool {
 	return config.AccountID != "" && config.APIKey != ""
 }
 
+func (config *NewRelicConfig) url() string {
```
[the remark this answers is not in the document]
Comment (shyamz-22) [reply atc/metric/emitter/newrelic.go]: I have made the necessary changes [here](https://github.com/shyamz-22/concourse/blob/2b635f22ecde5a676bd16e8e3935182158f25846/atc/metric/emitter/newrelic.go#L70) and removed the test that is not relevant anymore

**Pull request merged**